src/overlays/actors/ovl_Bg_Hakugin_Post/z_bg_hakugin_post.c

```diff
@@ -21,103 +54,994 @@ const ActorInit Bg_Hakugin_Post_InitVars = {
+        unkStruct2->unk_1C = 0.90999997f - ((0.04f - unkStruct2->unk_00) * 26.31579f * 0.02f);
+        unkStruct2->unk_20.x = Rand_Next() >> 0x10;
+        unkStruct2->unk_20.y = Rand_Next() >> 0x10;
+        unkStruct2->unk_20.z = Rand_Next() >> 0x10;
+        unkStruct2->unk_26 = (Rand_Next() & 0x3FFF) - 0x1FFF;
+        unkStruct2->unk_28 = (Rand_Next() & 0x1FFF) - 0xFFF;
+        unkStruct2->unk_2A = (Rand_Next() & 0x1FFF) - 0xFFF;
+        unkStruct2->unk_2C = (Rand_Next() & 0x1F) + 0x4B;
+        unkStruct2->unk_2D = 2;
+    }
+
+    for (i = 0, val = 0; i < 22; i++, val += 0xBA2) {
+        temp_f20 = (Rand_ZeroOne() * 60.0f) + 220.0f;
+        spA0.x = (Math_SinS(val) * temp_f20) + spB8.x;
+        spA0.y = (((Rand_ZeroOne() * 1.2f) - 0.1f) * spE4) + spB8.y;
+        spA0.z = (Math_CosS(val) * temp_f20) + spB8.z;
+        func_800B0E48(globalCtx, &spA0, &D_801D15B0, &D_80A9D8EC, &D_80A9D8E4, &D_80A9D8E8,
+                      (Rand_Next() >> 0x1A) + 0x82, (Rand_Next() >> 0x1A) + 0x6E);
+    }
+
+    temp_unkStruct1 = func_80A9B32C(unkStruct, unkStruct1);
+    if (temp_unkStruct1 != NULL) {
+        temp_f6 = (this->unk_16C + func_80A9B2B8(unkStruct)) - 40.0f;
+        spB8.x = temp_unkStruct1->unk_14.x + this->dyna.actor.world.pos.x;
+        spB8.y = temp_unkStruct1->unk_14.y + this->unk_16C;
+        spB8.z = temp_unkStruct1->unk_14.z + this->dyna.actor.world.pos.z;
+
+        for (i = 65, val = 0; i < ARRAY_COUNT(unkStruct->unk_02A4); i++, val += 0x4A7) {
+            unkStruct2 = &unkStruct->unk_02A4[i];
+            unkStruct2->unk_00 = (Rand_ZeroOne() * 0.007f) + 0.002f;
+            temp_s1_3 = (Rand_Next() & 0x7FF) + val;
+            temp_f24 = Math_SinS(temp_s1_3);
+            temp_f28 = Math_CosS(temp_s1_3);
+            unkStruct2->unk_04.x = (temp_f24 * 225.5f) + spB8.x;
+            if (i & 1) {
+                unkStruct2->unk_04.y = (Rand_ZeroOne() * D_80A9D880[temp_unkStruct1->unk_00].unk_00) + spB8.y;
+            } else {
+                unkStruct2->unk_04.y = temp_f6 - (Rand_ZeroOne() * 350.0f);
+            }
+            unkStruct2->unk_04.z = (temp_f28 * 225.5f) + spB8.z;
+            unkStruct2->unk_10.x = Rand_ZeroOne() + (temp_f24 * 7.0f);
+            unkStruct2->unk_10.y = 0.0f;
+            unkStruct2->unk_10.z = Rand_ZeroOne() + (temp_f28 * 7.0f);
+            unkStruct2->unk_1C = 0.90999997f - ((0.04f - unkStruct2->unk_00) * 26.31579f * 0.075f);
+            unkStruct2->unk_20.x = Rand_Next() >> 0x10;
+            unkStruct2->unk_20.y = Rand_Next() >> 0x10;
+            unkStruct2->unk_20.z = Rand_Next() >> 0x10;
+            unkStruct2->unk_26 = (Rand_Next() & 0x1FFF) - 0xFFF;
+            unkStruct2->unk_28 = (Rand_Next() & 0x1FFF) - 0xFFF;
+            unkStruct2->unk_2A = (Rand_Next() & 0x1FFF) - 0xFFF;
+            unkStruct2->unk_2C = (Rand_Next() & 0x1F) + 0x41;
+            unkStruct2->unk_2D = 0;
+        }
+    }
+    unkStruct->unk_1924 = 1;
```

Comment:
This seems to be used as a boolean
```suggestion
    unkStruct->unk_1924 = true;
```